deeprankcore/utils/metrics.py

```diff
@@ -27,11 +27,11 @@ def __exit__(self, exception_type, exception, traceback):
         pass # pylint: disable=unnecessary-pass
 
```

Comment:
So your intention is to make the metrics exporter always take a directory path as input, but why isn't this an argument to the __init__ function?

Reply:
> @gcroci2 I originally intended children of the `MetricsExporter` class to be added to a `MetricsExporterCollection` object within the `Trainer` object. This way, a metrics exporter can keep a file or a tensorboard writer open and close it at the end of the with-clause. You could add the `save_all_metrics` to the `__exit__` method, if you want it to be called at the end of a training/testing run.
> 
> I don't understand which if-statement you mean, sorry!

I updated the PR and my comment above, please have a look :)

Reply:
Now it is :)